listenbrainz/webserver/static/js/src/UserArtists.tsx

```diff
@@ -0,0 +1,192 @@
+    if (url.searchParams.get("page")) {
+      this.currPage = Number(url.searchParams.get("page"));
+    } else {
+      this.currPage = 1;
+    }
+
+    try {
+      const data = await this.APIService.getUserStats(
+        user.name,
+        undefined,
+        undefined,
+        1
+      );
+      this.maxListens = data.payload.artists[0].listen_count;
+      this.totalPages = Math.ceil(
+        data.payload.total_artist_count / this.ROWS_PER_PAGE
+      );
+    } catch (error) {
+      this.handleError(error);
+    }
+    this.handlePageChange(this.currPage);
+  }
+
+  getData = async (
+    userName: string,
+    offset: number
+  ): Promise<UserArtistsResponse> => {
+    const data = await this.APIService.getUserStats(
+      userName,
+      undefined,
+      offset,
+      this.ROWS_PER_PAGE
+    );
+    return data;
+  };
+
+  processData = (
+    data: UserArtistsResponse,
+    offset: number
+  ): UserArtistsData => {
+    const result = data.payload.artists
+      .map((elem, idx: number) => {
+        return {
+          id: `${offset + idx + 1}. ${elem.artist_name}`,
+          count: elem.listen_count,
+        };
+      })
+      .reverse();
+    return result;
+  };
+
+  handlePageChange = async (page: number): Promise<void> => {
+    const { user } = this.props;
+    const offset = (page - 1) * this.ROWS_PER_PAGE;
+    try {
+      const data = await this.getData(user.name, offset);
+      this.currPage = page;
+      this.setState({ data: this.processData(data, offset) });
+    } catch (error) {
+      this.handleError(error);
+    }
+  };
+
+  handleError = (error: Error): void => {
+    // Error Boundaries don't catch errors in async code.
+    // Throwing an error in setState fixes this.
+    // This is a hacky solution but should be fixed with upcoming concurrent mode in React.
+    const { user } = this.props;
+
+    if (
+      error.name === "SyntaxError" &&
+      error.message === "Unexpected end of JSON input"
+    ) {
+      // If the above error is thrown, it means stats haven't been calculated, show an
+      // appropriate message to the user.
+      this.setState(() => {
+        throw new Error(
+          `Statistics for user: ${user.name} have not been calculated yet. Please try again later.`
+        );
+      });
+    }
+
+    this.setState(() => {
+      throw error;
+    });
+  };
+
+  render() {
+    const { data } = this.state;
+    const prevPage = this.currPage - 1;
+    const nextPage = this.currPage + 1;
+
+    return (
+      <div>
+        <div className="row">
+          <div className="col-md-12" style={{ height: "75em" }}>
+            <Bar data={data} maxValue={this.maxListens} />
+          </div>
+        </div>
+        <div className="row">
+          <div className="col-md-12">
+            <ul className="pager">
+              <li className={`previous ${!(prevPage > 0) ? "hidden" : ""}`}>
+                {/* eslint-disable-next-line */}
```

Comment:
why do we need this eslint-disable?

Reply:
Ideally we should not be using anchor tags with `onClick` handler, but changing it to buttons leads to lot of styling issues. It should be fixed with [LB-579](https://tickets.metabrainz.org/browse/LB-579).